src/plugins/dashboard/public/dashboard_actions/filters_notification_popover.test.tsx

```diff
@@ -59,6 +60,9 @@ describe('filters notification popover', () => {
     updateQuery = (query) => querySubject.next(query);
     const viewModeSubject = new BehaviorSubject<ViewMode>('view');
     updateViewMode = (viewMode) => viewModeSubject.next(viewMode);
+    let canEditUnifiedSearchValue = true;
+    const canEditUnifiedSearch = jest.fn().mockReturnValue(canEditUnifiedSearchValue);
+    updateCanEditUnifiedSearch = (value: boolean) => (canEditUnifiedSearchValue = value);
```

Comment:
I think this is a little bit over complicated :) Can't we just have `canEditUnifiedSearch` as a mock and then update its returned value as necessary?


```typescript
const canEditUnifiedSearch = jest.fn().mockReturnValue(true);
    
<...>
    
beforeEach(async () => {
    <...>
    api = {
        <...>,
        canEditUnifiedSearch,
    };
}
```

Then, when we need it to return `false` instead, we can do the following:

```typescript
canEditUnifiedSearch.mockReturnValueOnce(false);
```

Reply:
Yeah that's true. I changed the test to use your suggestion.